core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1432,6 +1439,26 @@ class DAGScheduler(
     }
   }
 
+  /**
+   * Responds to a worker being removed. This is called inside the event loop, so it assumes it can
+   * modify the scheduler's internal state. Use workerRemoved() to post a loss event from outside.
+   *
+   * We will assume that we've lost all shuffle blocks associated with the host if a worker is
```

Comment:
This usually happens when we just finished recovery, in this case there shouldn't be much shuffle output generated, even there were some shuffle output, it should not cost much to re-generate them.
Of course the most ideal strategy would be to keep this condition separated from other WorkerRemove messages, I'm just hesitate to import more complexity to gain the few performance improvement.